src/main/combatLogParser.ts

```diff
@@ -0,0 +1,434 @@
+
+        this.emit('DataFirstEvent', handler.wowFlavour);
+
+        this.resetLockTimeout();
+
+        this._handlerLock = handler;
+        return true;
+    }
+
+    /**
+     * Reset the lock/data timeout timer such that it doesn't fire when it
+     * isn't supposed to.
+     */
+    private resetLockTimeout(): void {
+        if (this._handlerLockTimeout) {
+            clearTimeout(this._handlerLockTimeout);
+        }
+
+        this._handlerLockTimeout = setTimeout(() => {
+            console.log(`[CombatLogParser] Unlocking path '${this._handlerLock?.path}' for exclusive event processing since no data received in ${this._options.dataTimeout / 1000} seconds.`)
+            this._handlerLock = undefined;
+            this.emit('DataTimeout', this._options.dataTimeout);
+        }, this._options.dataTimeout);
+    }
+
+    /**
+     * Find and return the most recent file that matches the combat log filename
+     * pattern.
+     */
+    private getLatestLog (pathSpec: string): string | undefined {
+        const globPath = path.join(pathSpec, 'WoWCombatLog*.txt');
+        const logs = glob.sync(globPath)
+            .map((name: string) => ({name, mtime: fs.statSync(name).mtime}))
+            .sort((A: any, B: any) => B.mtime.getTime() - A.mtime.getTime());
+
+        if (logs.length === 0) {
+            return;
+        }
+
+        return logs[0].name;
+    }
+
+    /**
+     * Monitor a file for new content and process combat log lines accordingly
+     * when they arrive.
+     */
+    private tailLogFile(handler: CombatLogMonitorHandlerType): void {
+        // Clear any old tail handler before creating the new instance
+        if (handler.tail) {
+            handler.tail.unwatch();
+        }
+
+        const tailHandler = new tail(handler.currentLogFile, this._tailOptions);
+        tailHandler
+            .on('line', (line: string) => {
+                if (!this.lockHandler(handler)) {
+                    return;
+                }
+
+                this.handleLogLine(handler.wowFlavour, line);
+            })
+            .on('error', (error: unknown) => {
+                console.error('[CombatLogParser] ERROR: ', error);
+            });
+
+        handler.tail = tailHandler;
+    }
+
+    /**
+     * Watch a directory for combat logs and restart the monitoring
+     * mechanism if there's a new one detected.
+     */
+    private watchLogDirectory(path: string): void {
+        const handler = this._handlers[path];
+
+        handler.watchInterval = setInterval(() => {
+            const latestLogFile = this.getLatestLog(path);
+
+            // Handle the case where there is no logs in the WoW log directory.
+            if (!latestLogFile) {
+                return;
+            }
+
+            const logFileChanged = (latestLogFile !== handler.currentLogFile);
+
+            if (!handler.currentLogFile || logFileChanged) {
```

Comment:
Don't think you need the (!handler.currentLogFile) part of this check - it's already caught by the logFileChanged variable

Reply:
Yeah, that makes sense, nice find!